### PR TITLE
Processing updates

### PR DESCRIPTION
**Pull request by theHenks**
Added functionality which was so far in the dataflow:

- `is_analysis_run` checks if the run is a analysis run
- `start_key` returns the starting FileKey for a given period and run
- `phy_livetime` returns the livetime for a given run and period

Also, I extended the `channelinfo` to be able to get the info for a given run and period by using the `start_key` as a reference and also to get a handy access to the `:geds` subsytem with 
``` julia
channelinfo(l200, (:p03, :r000, :cal))
channelinfo(l200, (:p03, :r000, :cal); system=:geds, processable=true)
```
This makes this whole process of getting filekeys etc way more handy and the user can just get the `channelinfo` for a given run much more human readable.

I will also move some of the `addprocs` functionality and try to start with some of the logging stuff. 
Maybe @oschulz, you could also share some ideas here. The missing features are here a way to first create Markdown logs and then write them. 
Then, a function `savelegendfig` would be nice which automatically generates figure names and saves a given figure.
And the option to save units in the `PropDicts` as discussed.

### CODE REVIEW

File: src/dataprod_config.jl [hunk 1]
```diff
@@ -144,6 +144,62 @@ function _resolve_partition_runs(data::LegendData, period::DataPeriod, runs::Abs
     end
 end
 
+"""
```
Comment (oschulz) [inline src/dataprod_config.jl:147]: Let's do

```julia
(period, run) in analysis_runs(data::LegendData)
```

instead.

File: src/dataprod_config.jl [hunk 2]
```diff
@@ -144,6 +144,62 @@ function _resolve_partition_runs(data::LegendData, period::DataPeriod, runs::Abs
...
+    elseif pydataprod_config(data).analysis_runs[Symbol(period)] == "all"
+        return true
+    else
+        return string(run) in pydataprod_config(data).analysis_runs[Symbol(period)]
+    end
+end
+export is_analysis_run
+
+
+function _get_runinfo(data::LegendData, period::DataPeriodLike, run::DataRunLike, category::DataCategoryLike, key::Symbol)
```
Comment (oschulz) [inline src/dataprod_config.jl:165]: Let's have an exported function

```
runinfo(data::LegendData)::StructArray
```

instead, similar to `channelinfo`.

File: src/dataprod_config.jl [hunk 3]
```diff
@@ -144,6 +144,62 @@ function _resolve_partition_runs(data::LegendData, period::DataPeriod, runs::Abs
...
+end
+
+const _cached_start_filekeys = LRU{Tuple{DataPeriod, DataRun, DataCategory}, FileKey}(maxsize = 30)
+
+"""
+    start_key(data::LegendData, period::DataPeriodLike, run::DataRunLike, category::DataCategoryLike)
+
+Get the starting filekey for `data` in `period`, `run`, `category`.
+"""
+function start_filekey(data::LegendData, period::DataPeriodLike, run::DataRunLike, category::DataCategoryLike)
```
Comment (oschulz) [inline src/dataprod_config.jl:183]: Should use `runinfo(data, period, run)` internally.

File: src/dataprod_config.jl [hunk 4]
```diff
@@ -144,6 +144,62 @@ function _resolve_partition_runs(data::LegendData, period::DataPeriod, runs::Abs
...
+        throw(ArgumentError("Invalid run $run for key $key in period $period"))
+    elseif !haskey(data.metadata.dataprod.runinfo[Symbol(period)][Symbol(run)], Symbol(category))
+        throw(ArgumentError("Invalid category $category for key $key in period $period and run $run"))
+    end
+    data.metadata.dataprod.runinfo[Symbol(period)][Symbol(run)][Symbol(category)][key]
+end
+
+const _cached_start_filekeys = LRU{Tuple{DataPeriod, DataRun, DataCategory}, FileKey}(maxsize = 30)
+
+"""
```
Comment (oschulz) [inline src/dataprod_config.jl:178]: This can just become

```
runinfo(data, perior, category).startkey
```

**Review by oschulz — CHANGES_REQUESTED** [review]

File: src/dataprod_config.jl (hunk 1 above)
Comment (theHenks) [reply src/dataprod_config.jl:147]: Fixed but returned a StructArray instead to make it simnilar to the data_partitions. Also ok?

Comment (theHenks) [reply src/dataprod_config.jl:147]:
```suggestion
function analysis_runs(data::LegendData)
    aruns = pydataprod_config(data).analysis_runs
    periods_and_runs = [
        let period = DataPeriod(string(p))
            map(run -> (period = period, run = run), _resolve_partition_runs(data, period, rs))
        end
        for (p,rs) in aruns
    ]
    flat_pr = vcat(periods_and_runs...)::Vector{@NamedTuple{period::DataPeriod, run::DataRun}}
    StructArray(flat_pr)
end
export analysis_runs
```

File: src/dataprod_config.jl (hunk 2 above)
Comment (theHenks) [reply src/dataprod_config.jl:165]: Here my suggestion: 


```suggestion
function runinfo(data::LegendData, runsel::RunCategorySelLike)::NamedTuple
    # unpack runsel
    period, run, category = runsel
    period, run = DataPeriod(period), DataRun(run)
    # check if run, period and category is available
    if !haskey(data.metadata.dataprod.runinfo, Symbol(period))
        throw(ArgumentError("Invalid period $period"))
    elseif !haskey(data.metadata.dataprod.runinfo[Symbol(period)], Symbol(run))
        throw(ArgumentError("Invalid run $run in period $period"))
    elseif !haskey(data.metadata.dataprod.runinfo[Symbol(period)][Symbol(run)], Symbol(category))
        throw(ArgumentError("Invalid category $category in period $period and run $run"))
    end
    (
        startkey = FileKey(data.name, period, run, category, Timestamp(data.metadata.dataprod.runinfo[Symbol(period)][Symbol(run)][Symbol(category)][:start_key])), 
        livetime =  if haskey(data.metadata.dataprod.runinfo[Symbol(period)][Symbol(run)][Symbol(category)], :livetime_in_s)
                        Unitful.Quantity(data.metadata.dataprod.runinfo[Symbol(period)][Symbol(run)][Symbol(category)][:livetime_in_s], u"s") 
                    else 
                        Unitful.Quantity(NaN, u"s")
                    end
    )
end
export runinfo
```

File: src/legend_data.jl [hunk 5]
```diff
@@ -216,6 +220,8 @@ end
 
 """
     channelinfo(data::LegendData, sel::AnyValiditySelection)
```
Comment (oschulz) [inline src/legend_data.jl:222]: Let's do

```julia
channelinfo(data::LegendData, sel::AnyValiditySelection; system::Symbol = :all, only_processable::Bool = false)
channelinfo(data::LegendData, sel::RunSelLike; system::Symbol = :all, only_processable::Bool = false)
```

**Comment by theHenks** [comment]
Ready to merge now from my side. What do you say @oschulz ?

**Comment by oschulz** [comment]
Thanks!

**Pull request merged**